src/EditorFeatures/CSharpTest/Diagnostics/AddExplicitCast/AddExplicitCastTests_FixAllTests.cs

```diff
@@ -0,0 +1,1551 @@
+	void Foo2(Base2 b) { }
+
+	void Foo3(Derived2 b1) { }
+	void Foo3(int i) { }
+
+	private void M2(Base1 b1, Base2 b2, Base3 b3, Derived1 d1, Derived2 d2)
+	{
+		Foo1(b1);
+		Foo1(d1);
+		Foo2(b1);
+		Foo3(b1);
+	}
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+public class Program1
+{
+	class Base { }
+	class Derived : Base { }
+
+	class Test
+	{
+		private Derived d;
+		private Base b;
+		public Test(Derived derived)
+		{
+			d = derived;
+			B = derived;
+		}
+
+		public Derived D { get => d; set => d = value; }
+		public Base B { get => b; set => b = value; }
+
+		public void testing(Derived d) { }
+		private void testing(Base b) { }
+	}
+
+	class Test2 : Test
+	{
+		public Test2(Base b) : base((Derived)b) { }
+	}
+
+	class Test3
+	{
+		public Test3(Derived b) { }
+		public Test3(int i, Base b) : this((Derived)b) { }
+
+		public void testing(int i, Derived d) { }
+		private void testing(int i, Base d) { }
+	}
+
+	Base ReturnBase()
+	{
+		Base b = new Base();
+		return b;
+	}
+
+	void PassDerived(Derived d) { }
+	void PassDerived(int i, Derived d) { }
+
+	public Program1()
+	{
+		Base b;
+		Derived d = b;
+
+		PassDerived((Derived)b);
+		PassDerived((Derived)ReturnBase());
+		PassDerived(1, (Derived)b);
+		PassDerived(1, (Derived)ReturnBase());
+
+		<![CDATA[ List<Derived> list = new List<Derived>(); ]]>
+		list.Add((Derived)b);
+
+		Test t = new Test();
+		t.testing((Derived)b);
+
+		<![CDATA[ Func<Derived, Base> foo2 = d => d; ]]>
+		Derived d2 = foo2((Derived)b);
+		d2 = foo2(d);
+	}
+}
+        </Document>
+        <Document>
+public class Program2
+{
+	interface Base1 { }
+	interface Base2 : Base1 { }
+	interface Base3 { }
+	class Derived1 : Base2, Base3 { }
+	class Derived2 : Derived1 { }
+
+	void Foo1(Derived2 b) { }
+	void Foo2(Base2 b) { }
+
+	void Foo3(Derived2 b1) { }
+	void Foo3(int i) { }
+
+	private void M2(Base1 b1, Base2 b2, Base3 b3, Derived1 d1, Derived2 d2)
+	{
+		Foo1((Derived2)b1);
+		Foo1((Derived2)d1);
+		Foo2((Base2)b1);
+		Foo3((Derived2)b1);
+	}
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" AssemblyName=""Assembly2"" CommonReferences=""true"">
+        <Document>
+class Program3
+{
+	interface Base1 { }
+	interface Base2 : Base1 { }
+	interface Base3 { }
+	class Derived1 : Base2, Base3 { }
+	class Derived2 : Derived1 { }
+
+	void Foo1(Derived2 b) { }
+	void Foo2(Base2 b) { }
+
+	void Foo3(Derived2 b1) { }
+	void Foo3(int i) { }
+
+	private void M2(Base1 b1, Base2 b2, Base3 b3, Derived1 d1, Derived2 d2)
+	{
+		Foo1((Derived2)b1);
+		Foo1((Derived2)d1);
+		Foo2((Base2)b1);
+		Foo3((Derived2)b1);
+	}
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            await TestInRegularAndScriptAsync(input, expected);
+        }
+        #endregion
+    }
+}
```

Comment:
are there tests taht validate multiple fixes in a single expression?

Reply:
Multiple fixes cannot be fixed in one time if they are in a single expression, I can still add tests and see if they are fixed one by one

Reply:
should still have a test to validate :)